contracts/pool/pools/liquid/WithdrawalEligibility.sol

```diff
@@ -0,0 +1,163 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.18;
+
+import "../TokenPoolBase.sol";
+
+error UnlockPeriodNotFinished();
+error NonWithdrawableTokens();
+error AmountExceedsAvailableWithdraw();
+error StakedAmountExceeded();
+error LockedAmountExceeded();
+
+/**
+ * @title WithdrawalEligibility
+ * @dev Provides functionality for checking withdrawal eligibility and grace periods
+ */
+contract WithdrawalEligibility is TokenPoolBase {
+    uint256 internal constant GRACE_PERIOD = 4 weeks;
+    uint256 internal constant GRACE_PERIOD_REMOVAL_FEE = 1000;
+    uint256 internal constant UPGRADE_PERIOD = 63_158_400; // 2 years considering leap years in seconds;
+
+    struct EligibleWithdraw {
+        uint256 gracePeriodEndTime;
+        uint256 amount;
+    }
+
+    struct Stake {
+        EligibleWithdraw eligibleWithdraw;
```

Comment:
Mislim da se ovaj podatak trenutno ne moze dohvatit izvan contracta od nekog usera u sklopu mappinga `WithdrawalEligibility::stakes`, pa samo provjeri je li nam to okej.

Reply:
koliko sam razumia njima je bitno u kojem su poolu i tieru, tj da imamo dovoljno info sa contracta za kategorizirat ih i prikazat na UI. Za ovo nam je dovoljno ovo sta vec postoji di imamo getPoolStats koji nam vraca te podatke. Sada nisan sig jedino jesan krivo skuzia. @mmarinovic @mradmilovic